File: DailyChallenge/2020/March/7.java
// https://leetcode.com/explore/challenge/card/march-leetcoding-challenge-2021/588/week-1-march-1st-march-7th/3663/

class MyHashMap {

    /** Initialize your data structure here. */
    Integer[][] map;
    int NUM_OF_BUCKETS = 1000;
    int BUCKET_SIZE = 1000;
    public MyHashMap() {
        map = new Integer[NUM_OF_BUCKETS][BUCKET_SIZE];
    }

    /** value will always be non-negative. */
    public void put(int key, int value) {
        Integer[] bucket = getBucket(key);
        int keyHash = getKeyHash(key);
        bucket[keyHash] = value;
    }

    /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
    public int get(int key) {
        Integer[] bucket = getBucket(key);
        int keyHash = getKeyHash(key);
        return bucket[keyHash] == null ? -1 : bucket[keyHash];
    }

    /** Removes the mapping of the specified value key if this map contains a mapping for the key */
    public void remove(int key) {
        Integer[] bucket = getBucket(key);
        int keyHash = getKeyHash(key);
        bucket[keyHash] = null;
    }

    private Integer[] getBucket(int key) {
        int bucketIdx = key / BUCKET_SIZE;
        return map[bucketIdx];
    }

    private Integer getKeyHash(int key) {
        return key % BUCKET_SIZE;
    }
}

File: DailyChallenge/2020/June/7.py
# https://leetcode.com/explore/challenge/card/june-leetcoding-challenge-2021/603/week-1-june-1st-june-7th/3770/



class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        length=len(cost)
        if length==0 or length==1:
            return 0
        elif length==2:
            return min(cost[0],cost[1])
        count=[0 for i in range(0,length+1)]
        count[0]=0
        count[1]=0
        for i in  range(2,length+1):
            count[i]=min(count[i-2]+cost[i-2],count[i-1]+cost[i-1])
        return count[length]
